ui/ducks/send/send.js

```diff
@@ -705,12 +728,85 @@ const slice = createSlice({
       slice.caseReducers.calculateGasTotal(state);
     },
     /**
-     * sets the provided gasPrice in state and then recomputes the gasTotal
+     * Sets the appropriate gas fees in state and determines and sets the
+     * appropriate transactionType based on gas fee fields received.
      */
-    updateGasPrice: (state, action) => {
-      state.gas.gasPrice = addHexPrefix(action.payload);
+    updateGasFees: (state, action) => {
+      if (
+        action.payload.transactionType === TRANSACTION_ENVELOPE_TYPES.FEE_MARKET
+      ) {
```

Comment:
Question for me to figure out: under what conditions can we hit this case? (It is a question because if `action.payload.transactionType === TRANSACTION_ENVELOPE_TYPES.FEE_MARKET`, then I assume we are are on EIP1559 compatible network, in which case I thought that we could _not_ update gas feels while on the send screen)... maybe (probably?) this is just for updates that happen in the background, as the user sets token and recipient?

Reply:
- [ ] Understand this

Reply:
> maybe (probably?) this is just for updates that happen in the background, as the user sets token and recipient?

This is correct. Also, I believe it best that we do not update gas fee properties while in edit mode/stage. I have a change for that locally and will add it to this PR